### PR TITLE
update samples for interfaces

### PR DESCRIPTION
**Pull request by BillWagner**
contributes to dotnet/docs#12490

This PR updates the samples for the reference page for interfaces.

I'll make a cohesive set of samples for interfaces when I update the programming guide section on interfaces.

### CODE REVIEW

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 1]
```diff
@@ -98,46 +98,39 @@ int y
...
        public Point(int x, int y)
        {
-          _x = x;
-          _y = y;
+          X = x;
+          Y = y;
        }
 
        // Property implementation:
        public int x
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
       public int X { get; set; }
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 2]
```diff
@@ -98,46 +98,39 @@ int y
...
        {
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 3]
```diff
@@ -98,46 +98,39 @@ int y
...
-          get
-          {
-             return _x;
-          }
-
-          set
-          {
-             _x = value;
-          }
+          get; // Use auto-implemented properties
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 4]
```diff
@@ -98,46 +98,39 @@ int y
...
+          set;
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 5]
```diff
@@ -98,46 +98,39 @@ int y
...
        }
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 6]
```diff
@@ -98,46 +98,39 @@ int y
...
 
        public int y
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
       public int Y { get; set; }
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 7]
```diff
@@ -98,46 +98,39 @@ int y
...
        {
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 8]
```diff
@@ -98,46 +98,39 @@ int y
...
-          get
-          {
-             return _y;
-          }
-          set
-          {
-             _y = value;
-          }
+          get;
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 9]
```diff
@@ -98,46 +98,39 @@ int y
...
+          set;
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 10]
```diff
@@ -98,46 +98,39 @@ int y
...
        }
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]:
```suggestion
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 11]
```diff
@@ -98,46 +98,39 @@ int y
...
           set;
        }
+
+       public double Distance 
+       {
+           get;
+       }
     }
 
     class Point : IPoint
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]: The changes requesting to rename the properties from lowercase to uppercase should be done in the interface too.

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 12]
```diff
@@ -98,46 +98,39 @@ int y
           get;
           set;
        }
+
+       public double Distance
```
Comment (Youssef1313) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]: I think access modifiers shouldn't be specified in an interface.

```suggestion
       double Distance 
```

File: snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs [hunk 13]
```diff
@@ -87,64 +87,49 @@ static void Main()
     interface IPoint
     {
        // Property signatures:
-       int x
+       int X
```
Comment (gewarren) [inline snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]: Isn't this an instance auto-implemented property? I thought this wasn't allowed.

Comment (BillWagner) [reply snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/keywordsTypes.cs]: In an interface definition, this defines the contract, it doesn't define an implementation. 

The sample is correct. I'll update the article text to make that more clear. Good catch!

**Pull request merged**